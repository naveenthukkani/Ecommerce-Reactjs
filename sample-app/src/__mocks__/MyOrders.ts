import {Orders} from "../../src/myorders/Orders";

export const OrdersList: Array<Orders> = [
        {
            date: new Date().toLocaleString(),
            status: "Delivered",
            isbn: "9781593275846",
            title: "Eloquent JavaScript, Second Edition",
            subtitle: "A Modern Introduction to Programming",
            author: "Marijn Haverbeke",
            pages: 472,
            price: 500,
            id:1
        },
];