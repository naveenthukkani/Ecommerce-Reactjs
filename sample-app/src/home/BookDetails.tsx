import React, {Component} from 'react';
import { connect } from 'react-redux';
import {RouteComponentProps} from 'react-router-dom'
import BookImage from '../images/book.jpg'
import {actions, selectors} from '../store'
import {Book} from "./Book"
import { Order } from '../myorders/Order';

interface ComponentProps extends RouteComponentProps{
  id: number,
}

interface MapStateToPropsTypes {
  listOfBooks: Book[]
}
class BookDetails extends Component <any>{
  constructor(props:any){
    super(props);
  }

  buyButtonAction = (e:any,book: Book) => {
    const orderObj :Order= {
      "date": new Date().toLocaleString(),
      "status": "Delivered",
      "title": book?.title,
      "subtitle": "",
      "author": book?.author,
      "price": book?.price,
      "id":book?.id,
      "pages": book?.pages,
      "isbn": book?.isbn
    } 
    alert(`${book.title} book checkout succefully `)
    e.stopPropagation();
    this.props.actions.buyBook(orderObj);
  }

  addToCartAction = (e: any , book: Book) =>{
    alert(`${book.title} book added to cart succefully `)
    e.stopPropagation();
    this.props.actions.addToCart(book);
  }

  render(){
    const book = this.props.listOfBooks.find((obj: any) =>{ 
      return obj.id ===Number(this.props.match.params.id)
    });
    
    return (
      <div>
        {book ? <div className="Book-details-container">
            <div className="Image-container">
                <img src={BookImage} alt="Book"></img>
            </div>
            <div className="Details-container">
            <h1>{book?.title}</h1>
            <div>
              <label>Book Price: </label>
              <span>{book?.price}</span>
            </div>
            <div>
              <label>Author Name: </label>
              <span>{book?.author}</span>
            </div>
            <div>
              <label>Page Count: </label>
              <span>{book?.pages}</span>
            </div>
            <div>
              <label>ISBN: </label>
              <span>{book?.isbn}</span>
            </div>
            <div>
            <button className="details-buttons" onClick={(e) => this.addToCartAction(e,book)}>Add to cart</button>
            <button className="details-buttons" onClick={(e) => this.buyButtonAction(e,book)}>Buy Now</button>
            </div>
            <div className="Book-Description-detail">
              <label>Description: </label>
              <p>{book?.description}</p>
            </div>
            </div>
            </div> :""}
        </div>
    )
  }
}
const mapDispatchToProps=(dispatch: any)=> ({
  actions: { 
    buyBook: (orderObj: Order) => dispatch(actions.myorders.fetchRequest(orderObj)),
    addToCart: (book: Book)=> dispatch(actions.cart.fetchRequest(book))
  }
});
const mapStateToProps = (state: any) => ({
  listOfBooks: selectors.home.getBooksList(state)
});

export {BookDetails}
export default connect<MapStateToPropsTypes,any>(
  mapStateToProps,
  mapDispatchToProps
)(BookDetails);