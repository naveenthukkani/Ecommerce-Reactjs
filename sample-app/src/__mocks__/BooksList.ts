import {Books} from "../../src/home/Books";

export const BooksList: Array<Books> = [
   {
       name:"maths",
       discription: "it is simple problems book for children",
       author:"naveen",
       price: 5,
       pagecount: 1,
	   isbn:"",
	   id:1,
   },
   {
    name:"social",
    discription: "it is sample book for children",
    author:"naveen",
    price: 7,
    pagecount: 1,
	isbn:"",
	id:2,
},
{
    name:"telugu",
    discription: "it is language learning book",
    author:"naveen",
    price: 8,
    pagecount: 1,
	isbn:"",
	id:3,
},
{
    name:"English",
    discription: "it is international language learning book",
    author:"naveen",
    price: 5,
    pagecount: 1,
	isbn:"",
	id:4,
},
{
    name:"maths",
    discription: "it is simple problems book for children",
    author:"naveen",
    price: 5,
    pagecount: 1,
	isbn:"",
	id:5,
},
{
	name:"social",
	discription: "it is sample book for children",
	author:"naveen",
	price: 7,
	pagecount: 1,
	isbn:"",
	id:6,
},
{
	name:"telugu",
	discription: "it is language learning book",
	author:"naveen",
	price: 8,
	pagecount: 1,
	isbn:"",
	id:7,
},
{
	name:"English",
	discription: "it is international language learning book",
	author:"naveen",
	price: 5,
	pagecount: 1,
	isbn:"",
	id:8,
},
{
	name:"social",
	discription: "it is sample book for children",
	author:"naveen",
	price: 7,
	pagecount: 1,
	isbn:"",
	id:9,
},
{
	name:"telugu",
	discription: "it is language learning book",
	author:"naveen",
	price: 8,
	pagecount: 1,
	isbn:"",
	id:10,
},
{
	name:"English",
	discription: "it is international language learning book",
	author:"naveen",
	price: 5,
	pagecount: 1,
	isbn:"",
	id:11,
}
]