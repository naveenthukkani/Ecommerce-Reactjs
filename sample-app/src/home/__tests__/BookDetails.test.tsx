import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {BookDetails} from '../BookDetails';
import { BooksList } from '../../__mocks__/BooksList';

describe('BookDetails', () =>{
    let props:any;
    beforeEach(()=>{
        props= {
         listOfBooks: BooksList
        }
    });

    it('renders the component correctly', () => {
        const component = shallow(<BookDetails {...props}  match={{params: {id: 1}}} />);
        expect(toJson(component)).toMatchSnapshot();
      });
});

