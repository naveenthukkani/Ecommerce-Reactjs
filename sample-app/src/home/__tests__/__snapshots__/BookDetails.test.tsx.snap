// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetails renders the component correctly 1`] = `
<div>
  <div
    className="Book-details-container"
  >
    <div
      className="Image-container"
    >
      <img
        alt="Book"
        src="book.jpg"
      />
    </div>
    <div
      className="Details-container"
    >
      <h1>
        Eloquent JavaScript, Second Edition
      </h1>
      <div>
        <label>
          Book Price: 
        </label>
        <span>
          500
        </span>
      </div>
      <div>
        <label>
          Author Name: 
        </label>
        <span>
          Marijn Haverbeke
        </span>
      </div>
      <div>
        <label>
          Page Count: 
        </label>
        <span>
          472
        </span>
      </div>
      <div>
        <label>
          ISBN: 
        </label>
        <span>
          9781593275846
        </span>
      </div>
      <div>
        <button
          className="details-buttons"
          onClick={[Function]}
        >
          Add to cart
        </button>
        <button
          className="details-buttons"
          onClick={[Function]}
        >
          Buy Now
        </button>
      </div>
      <div
        className="Book-Description-detail"
      >
        <label>
          Description: 
        </label>
        <p>
          JavaScript lies at the heart of almost every modern web application, from social apps to the newest browser-based games. Though simple for beginners to pick up and play with, JavaScript is a flexible, complex language that you can use to build full-scale applications.
        </p>
      </div>
    </div>
  </div>
</div>
`;
